test/unit/models/publishes_to_publishing_api_test.rb

```diff
@@ -2,85 +2,68 @@
-    Whitehall::PublishingApi.expects(:publish_async).with(organisation)
-    organisation.update_attribute(:name, 'Edited org')
+  setup do
+    TestObj.stubs(:after_commit).with(:publish_to_publishing_api,
+                                      {if: :can_publish_to_publishing_api?})
+    TestObj.stubs(:after_commit).with(:publish_gone_to_publishing_api,
+                                      {on: :destroy})
   end
 
-  test "create does not publish to Publishing API if disallowed" do
-    organisation = build(:organisation)
-    organisation.stubs(:can_publish_to_publishing_api?).returns(false)
-    Whitehall::PublishingApi.expects(:publish_async).never
-    organisation.save
+  test "it hooks up publish_to_publishing_api correctly" do
+    TestObj.expects(:after_commit).with(:publish_to_publishing_api,
+                                         {if: :can_publish_to_publishing_api?})
+    include_module(TestObj.new)
   end
 
-  test "update does not publish to Publishing API if disallowed" do
-    organisation = create(:organisation)
-    organisation.stubs(:can_publish_to_publishing_api?).returns(false)
-    Whitehall::PublishingApi.expects(:publish_async).never
-    organisation.update_attribute(:name, 'Edited org')
+  test "it hooks up publish_gone_to_publishing_api correctly" do
+    TestObj.expects(:after_commit).with(:publish_gone_to_publishing_api,
+                                         {on: :destroy})
+    include_module(TestObj.new)
   end
 
-  test "update publishes to Publishing API using :en locale when no translated fields are set" do
-    person = create(:person, attributes_for(:person).except(:biography))
-
-    content_item = PublishingApiPresenters.presenter_for(person)
-    requests = [
-      stub_publishing_api_put_content(content_item.content_id, content_item.content),
-      stub_publishing_api_put_links(content_item.content_id, links: content_item.links),
-      stub_publishing_api_publish(content_item.content_id, locale: content_item.content[:locale], update_type: 'major')
-    ]
-
-    assert_all_requested(requests)
+  test "can_publish_to_publishing_api_returns_true_when_object_persisted" do
+    t_obj = include_module(TestObj.new)
+    assert t_obj.can_publish_to_publishing_api?
   end
 
-  test "update publishes to Publishing API using :en locale when the object is not translatable" do
-    policy_group = create(:policy_group)
-
-    content_item = PublishingApiPresenters.presenter_for(policy_group)
-    requests = [
-      stub_publishing_api_put_content(content_item.content_id, content_item.content),
-      stub_publishing_api_put_links(content_item.content_id, links: content_item.links),
-      stub_publishing_api_publish(content_item.content_id, locale: content_item.content[:locale], update_type: 'major')
-    ]
+  test "can_publish_to_publishing_api_returns_false_when_object_not_persisted" do
+    t_obj = include_module(TestObj.new)
+    t_obj.stubs(:persisted?).returns(false)
```

Comment:
`stubs` can be used as so `stubs(persisted?: false)` so you can drop the `returns` message for conciseness.